website/server/controllers/api-v3/tasks.js

```diff
@@ -901,11 +901,19 @@ api.clearCompletedTodos = {
       userId: user._id,
       type: 'todo',
       completed: true,
-      $or: [
-        {'challenge.id': {$exists: false}},
-        {'challenge.broken': {$exists: true}},
-        {'group.id': {$exists: false}},
-        {'group.broken': {$exists: true}},
+      $and: [ // exclude challenge and group tasks
```

Comment:
I think that this outer $and might  not be necessary since $and is used by default when multiple conditions are given. Just check that though before taking my word for it. :)